File: first_100_primes/c++/solution.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(int argc, char** argv)
{
	vector<bool> primes(1000, true);
	primes[0] = false;
	primes[1] = false;
	for (int i = 2; i < 100; i++)
	{
		for (int j = i * 2; j < 1000; j += i)
		{
			primes[j] = false;
		}
	}
	cout << 2;
	int curPrime = 3;
	for (int i = 1; i < 100; i++)
	{
		while (!primes[curPrime])
			curPrime++;

		cout << ',' << curPrime;
		curPrime++;
	}
}

File: first_100_primes/go/solution.go
package main

import (
	"fmt"
)

func main() {
	primes := make([]bool, 1000)
	primes[0] = false;
	primes[1] = false;
	for i := 2; i < len(primes); i++ {
		primes[i] = true;
	}

	for i := 2; i < 100; i++ {
		for j := i * 2; j < 1000; j += i {
			primes[j] = false;
		}
	}

	fmt.Printf("2");
	curPrime := 3;

	for i := 1; i < 100; i++ {
		for !primes[curPrime] {
			curPrime++
		}
		fmt.Printf(",%d", curPrime)
		curPrime++
	}
}
